pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -119,12 +165,21 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 			if port.Protocol == protocol.UDP {
 				continue
 			}
-			lbEndpoints := cb.buildLocalityLbEndpoints(networkView, service, port.Port, nil)
+			clusterKey := buildClusterKey(service, port, cb)
+			cached, tokens, allFound := cb.getAllCachedSubsetClusters(*clusterKey)
+			if allFound && !features.EnableUnsafeAssertions {
+				hit += len(cached)
+				resources = append(resources, cached...)
+				continue
+			} else {
+				miss += len(cached)
+			}
+
+			lbEndpoints := cb.buildLocalityLbEndpoints(clusterKey.networkView, service, port.Port, nil)
```

Comment:
nit: Can you add a comment here that we did not find cluster in cache so building now?